src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
@mDuo13 made an interesting observation recently in #cpp-informal that the sell flag is meaningless once an order is placed on to the books, as at that point it's not possible to get a better rate. 

Again, I wonder if the {asks, bids, flipped: {asks, bids}} grouping is really ideal, and whether the self flag is a great basis for orienting the quantity to either of takerGets and takerPaid. 

With a more consistent orientation, that's not dependent on the sell flag, the makerExchangeRate would also have a consistent relationship with the quantity and totalPrice fields.

> also exactly what the new API is trying to do

Whether it's a buy or sell type, the amount is always, the amount of BTC.  I guess that's equivalent to the `quantity` being whatever is being sold or bought.

Still, if the sell flag is ultimately meaningless, for orders in a book, I wonder what *is* meaningful? To me it's that the maker is wanting to exchange A for B. 

I'm trying to keep an open mind, but this still feels like an unnatural superimposition (or as I put it before, "trying to smooth/wash over something that's just fundamentally different" )